### PR TITLE
Ignore case in `$collection->remove()`

### LINKED ISSUE #5704: `Kirby\Toolkit\Collection::__unset()` does not remove item, when cases don’t match
## Description

[`Kirby\Toolkit\Collection::__unset()`](https://github.com/getkirby/kirby/blob/3.9.6.1/src/Toolkit/Collection.php#L124) does not remove item, when cases don’t match. 
However [`Kirby\Toolkit\Collection::__set()`](https://github.com/getkirby/kirby/blob/3.9.6.1/src/Toolkit/Collection.php#L100) updates the item, even when the cases don’t match.

**Expected behavior**  
`__set()` and `__unset()` should behave the same way.

**Possible solution**
```php
// src/Toolkit/Collection.php#L100
	/**
	 * Low-level element remover
	 *
	 * @param mixed $key the name of the key
	 */
	public function __unset($key)
	{
		if ($this->caseSensitive !== true) {
			$key = strtolower($key);
		}

		unset($this->data[$key]);
	}
```

**Additional context**
This caused an issue with [Merx](https://github.com/wagnerwagner/merx), where `Collection` is used for `Cart`. In some cases page uuids (which are mixed-case, e.g. `page://yEWQBkKPvrwBQfBw`) where used as keys. The keys are transformed to lower-case (e.g. `page://yewqbkkpvrwbqfbw`), however in some cases [`$cart->remove()`](https://github.com/wagnerwagner/merx/blob/v1.7.2/src/Cart.php#L87) is used with mixed-cased key.  
I created a [workaround](https://github.com/wagnerwagner/merx/commit/82db3b0473ac907405efc2aa5267a4e909940b01) for Merx.

See https://github.com/wagnerwagner/merx-starterkit/issues/11, https://github.com/wagnerwagner/merx-starterkit/issues/10

## To reproduce

Run the code

```php
<?php

use Kirby\Toolkit\Collection;

require __DIR__ . '/kirby/bootstrap.php';

$kirby = new Kirby();

$collection = new Collection();

$collection->__set('lorem', 'data');
dump($collection->toJson()); // {"lorem":"data"}

$collection->__set('lOrEM', 'newData');
dump($collection->toJson()); // {"lorem":"newData"}

$collection->__unset('lOrEM');
dump($collection->toJson()); // {"lorem":"newData"} – should be []
```

## Your setup
**Kirby Version**  
[3.9.6.1](https://github.com/getkirby/kirby/releases/tag/3.9.6.1), relevant code did not change in 4.0.

**Console output**  
–

**Your system (please complete the following information)**
 - Device: MacBook
 - OS: macOS
 - Browser: Safari
 - Version: 16.6

**Additional context**  
–

### PR DESCRIPTION
**Pull request by lukasbestle**
### Fixes

- `$collection->remove()` and `$collection->__unset()` in Toolkit collections behave like `$collection->set()`/`$collection->__set()` by default and ignore the key case

### Breaking changes

None

## Ready?

- [x] Unit tests for fixed bug/feature
- [x] In-code documentation (wherever needed)
- [x] Tests and checks all pass

### For review team

- [x] Add changes to release notes draft in Notion
- [x] ~Add to [website docs release checklist](https://github.com/getkirby/getkirby.com/pulls) (if needed)~

### CODE REVIEW

File: src/Cms/Collection.php [hunk 1]
```diff
@@ -79,6 +79,14 @@ public function __set(string $id, $object): void
 		$this->data[$id] = $object;
 	}
 
+	/**
+	 * Internal remover for each object in the Collection
```
Comment (distantnative) [inline src/Cms/Collection.php]: Should we add a comment on these methods here, why we're overwriting the toolkit class methods?

**Review by distantnative — APPROVED** [review]

**Comment by lukasbestle** [comment]
@bastianallgeier This is now also in 3.9.7 already, isn't it?

**Comment by lukasbestle** [comment]
I've branched off a `release/3.9.7` branch that doesn't include this PR, so we can release 3.9.7 with the safe RC-tested contents only (except the other PR we accidentally merged that does not have a regression risk in contrast to this one).

**Comment by bastianallgeier** [comment]
Argh, sorry. I was in the v4 zone and didn't see the milestone. I think it's better to release the release branch without this change.

**Pull request merged**